federation-js/src/types.ts

```diff
@@ -13,6 +13,17 @@ import {
   isObjectType,
 } from 'graphql';
 import { PromiseOrValue } from 'graphql/jsutils/PromiseOrValue';
+import { CacheHint, CachePolicy } from 'apollo-server-types';
+
+declare module 'graphql/type/definition' {
```

Comment:
This is probably not required any more once you rebase on https://github.com/apollographql/federation/pull/875 ?

Reply:
This still seems to be necessary even after a rebase

Reply:
My worry is that if you end up loading both this and apollo-server-core in your project, TypeScript will be sad to see the same declaration twice.

I'm thinking of making a chance so that the type under `cacheControl` is exported from apollo-server-types, so you can do something like:

```
  if ((info as any).cacheControl?.cacheHintFromType) {
    const cacheControl: ResolveInfoCacheControl = info.cacheControl;
    const cacheHint = cacheControl.cacheHintFromType(type);
```